board/safety.h

```diff
@@ -67,6 +67,71 @@ bool msg_allowed(int addr, int bus, const AddrBus addr_list[], int len) {
   return allowed;
 }
 
+// compute the time elapsed (in microseconds) from 2 counter samples
+// case where ts < ts_last is ok: overflow is properly re-casted into uint32_t
+uint32_t get_ts_elapsed(uint32_t ts, uint32_t ts_last) {
+  return ts - ts_last;
+}
+
+int get_addr_check_index(CAN_FIFOMailBox_TypeDef *to_push, AddrCheckStruct addr_list[], const int len) {
+  int bus = GET_BUS(to_push);
+  int addr = GET_ADDR(to_push);
+
+  int index = -1;
+  for (int i = 0; i < len; i++) {
+    for (uint8_t j = 0U; j < addr_list[i].addr_len; j++) {
+      // TODO: remove rule to skip bus check if expected bus is negative. See safety_honda.h
+      if ((addr == addr_list[i].addr[j]) && ((bus == addr_list[i].bus) || (addr_list[i].bus < 0))) {
+        index = i;
+        i = len;  // break also second loop
+        break;
+      }
+    }
+  }
+  return index;
+}
+
+void check_lagging_addrs(const safety_hooks *hooks) {
+  uint32_t ts = TIM2->CNT;
+  for (int i=0; i < hooks->addr_check_len; i++) {
+    uint32_t elapsed_time = get_ts_elapsed(ts, hooks->addr_check[i].last_timestamp);
+    // lag threshold is max of: 1s and MAX_MISSED_MSGS * expected timestep.
```

Comment:
The check is called at 1Hz, so the worst case time to trigger is 2 seconds. Should we document this?